react-hot-button-component/src/hot-button.jsx

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+
+class HotButton extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleClick = this.handleClick.bind(this);
+    this.state = {
+      clicks: 0,
+      className: 'hot-button cold'
+    };
+  }
+
+  handleClick() {
+    this.setState({ clicks: this.state.clicks + 1 });
+    if (this.state.clicks < 4) {
+      this.setState({ className: 'hot-button cold' });
+    } else if (this.state.clicks < 7) {
+      this.setState({ className: 'hot-button cool' });
+    } else if (this.state.clicks < 10) {
+      this.setState({ className: 'hot-button tepid' });
+    } else if (this.state.clicks < 13) {
+      this.setState({ className: 'hot-button warm' });
+    } else if (this.state.clicks < 16) {
+      this.setState({ className: 'hot-button hot' });
+    } else {
+      this.setState({ className: 'hot-button nuclear' });
+    }
+  }
```

Comment:
- Since `setState` is asynchronous, the `clicks` state won't be updated until _after_ `setState` is called with the new `className` state. It's generally a bad idea to call `setState` more than once in a single function call.
- Apps are generally easier to build and maintain (i.e. debug) when there's less stuff in state. Since the `className` here is entirely determined by the value of the `clicks` state, `className` should be calculated on-the-fly while rendering instead of living in the component state.